toolchain/check/facet_type.h

```diff
@@ -13,19 +13,21 @@
 
 namespace Carbon::Check {
 
-// Either an InterfaceId or a NamedConstraintId, used in places where they can
-// be used interchangeably, which is places that turn into facet types.
-using InterfaceOrNamedConstraintId =
-    std::variant<SemIR::InterfaceId, SemIR::NamedConstraintId>;
-
-// Create a FacetType typed instruction object consisting of a single
-// interface. The `specific_id` specifies arguments in the case the interface is
-// generic.
-auto FacetTypeFromInterface(Context& context, InterfaceOrNamedConstraintId id,
+// Create a FacetType typed instruction object consisting of a interface. The
+// `specific_id` specifies arguments in the case the interface is generic.
+//
+// The resulting FacetType may contain multiple interfaces if the named
+// interface contains `require` declarations.
+auto FacetTypeFromInterface(Context& context, SemIR::InterfaceId interface_id,
```

Comment:
Have you considered making these functions an overload set, e.g. `FacetTypeFromInterfaceOrNamedConstraint`? That would let you avoid the `if constexpr` at the callsite.